src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,85 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int COUNT_OF_FIGURE = 5;
+    private static final int MAX_VALUE_FOR_CALCULATIONS = 10;
+    private static final int DEFAULT_RADIUS_FOR_CIRCLE = 10;
+    private static final int SETTING_FOR_GENERATING_DOUBLE_NUMBER = 3;
+    private static final int SETTING_FOR_GENERATING_RANDOM_NUMBER_MINUS_ONE = 1;
+    private Random random = new Random();
+    private ColorSupplier randomColor = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        switch (getRandomNumberFigure()) {
+            case 1:
+                return getCircle();
+            case 2:
+                return getIsoscelesTrapezoid();
+            case 3:
+                return getRectangle();
+            case 4:
+                return getRightTriangle();
+            default:
+                return getSquare();
+        }
+    }
+
+    private int getRandomNumberFigure() {
+        return random.nextInt(COUNT_OF_FIGURE);
+    }
+
+    private Circle getCircle() {
+        double radius = SETTING_FOR_GENERATING_DOUBLE_NUMBER * random.nextDouble()
+                + random.nextInt(MAX_VALUE_FOR_CALCULATIONS)
+                - SETTING_FOR_GENERATING_RANDOM_NUMBER_MINUS_ONE;
```

Comment:
A big and complex expression, which is hard to read. Can we extract this to a separate method?

Or consider changing this logic. Applicable for all other cases.